### PR TITLE
elif

### PR DESCRIPTION
**Pull request by Preciseful**
added @elif
(not thoroughly tested though)

### CODE REVIEW

File: app/javascript/src/utils/compiler.js
Comment (CactusPuppy) [inline app/javascript/src/utils/compiler.js:1]: This diff is meaningless as it makes it appear that the entire file has been changed. Please ensure that the diff only includes meaningful changes and recommit.

**Review by CactusPuppy — CHANGES_REQUESTED** [review]

File: app/javascript/src/utils/compiler.js
Comment (netux) [reply app/javascript/src/utils/compiler.js:1]: image

But yes, you shouldn't change the styling of the file. Turn off autoformat on your IDE.

File: app/javascript/src/utils/compiler.js [hunk 1]
```diff
@@ -1,520 +1,570 @@
...
+
+  let match
+  while ((match = ifStartRegex.exec(joinedItems)) != null) {
+    const openingConditionParenIndex = match.index + match[0].length - 1
+    const closingConditionParenIndex = getClosingBracket(joinedItems, "(", ")", openingConditionParenIndex - 1)
+    if (closingConditionParenIndex < 0) {
+      continue
+    }
+
+    console.log("Getting if...")
```
Comment (CactusPuppy) [inline app/javascript/src/utils/compiler.js:276]: Please be sure to remove all console logging statements added for the purposes of debugging.

File: app/javascript/src/utils/compiler.js [hunk 2]
```diff
@@ -1,520 +1,570 @@
...
+    let conditionalEndingIndex = closingBracketIndex
+
+    let trueBlockContent = joinedItems.substring(openingBracketIndex + 1, closingBracketIndex)
+    let falseBlockContent = ""
+
+    let passed = evaluateExpressionTree(conditionExpressionTree)
+    elifStartRegex.lastIndex = closingBracketIndex
+    let elifMatch = elifStartRegex.exec(joinedItems)
+
+    console.log(`If passed with: ${ passed }`)
```
Comment (CactusPuppy) [inline app/javascript/src/utils/compiler.js:302]: Remove this debug console.log statement

File: app/javascript/src/utils/compiler.js [hunk 3]
```diff
@@ -1,520 +1,570 @@
...
+        falseBlockContent = joinedItems.substring(afterElseMatchedTextIndex, matchingClosingBracketForElseIndex)
+        conditionalEndingIndex = matchingClosingBracketForElseIndex
+      } else {
+        continue
+      }
+    }
+
+    console.log(`True block is: ${ trueBlockContent }`)
+    console.log(`False block is: ${ falseBlockContent }`)
+    console.log(`Match index is: ${ match.index } and ending is ${ conditionalEndingIndex + 1 }`)
```
Comment (CactusPuppy) [inline app/javascript/src/utils/compiler.js:358]: More debug console.log statements

File: app/javascript/src/utils/compiler.js [hunk 4]
```diff
@@ -1,520 +1,570 @@
...
+      conditionalEndingIndex = closingBracketIndex
+
+      if(!passed) {
+        passed = evaluateExpressionTree(conditionElifExpressionTree)
+        if(passed) {
+          trueBlockContent = joinedItems.substring(openingElifBracketIndex + 1, closingBracketIndex)
+          console.log("Else If passed with: true")
+        }
+        else
+          console.log("Else If passed with: false")
```
Comment (CactusPuppy) [inline app/javascript/src/utils/compiler.js:336]: More debug console.log statements

File: app/javascript/src/utils/compiler.js [hunk 5]
```diff
@@ -1,520 +1,570 @@
...
+    let falseBlockContent = ""
+
+    let passed = evaluateExpressionTree(conditionExpressionTree)
+    elifStartRegex.lastIndex = closingBracketIndex
+    let elifMatch = elifStartRegex.exec(joinedItems)
+
+    console.log(`If passed with: ${ passed }`)
+
+    while (elifMatch != null && elifMatch.index === closingBracketIndex + 1) {
+      console.log("Getting else if...")
```
Comment (CactusPuppy) [inline app/javascript/src/utils/compiler.js:305]: Be sure to remove this console.log statement

**Review by CactusPuppy — CHANGES_REQUESTED** [review]
Don't forget to remove your debug `console.log` statements before you commit!

File: app/javascript/src/utils/extraCompletions.js
Comment (netux) [inline app/javascript/src/utils/extraCompletions.js:1]: What is this for?

**Review by netux — CHANGES_REQUESTED** [review]
Nice solution! From my testing it seems to work fine.
I would have maybe tried splitting `evaluateConditionals` into multiple functions for handling `if` and `else`, since the function is getting a little convoluted and repetitive. But it's probably okay for now.

**Pull request closed without merging**